lib/filewatch/read_mode/processor.rb

```diff
@@ -103,10 +103,21 @@ def process_active(watched_files)
         else
           read_file(watched_file)
         end
+        
+        if @settings.exit_after_read
+          common_detach_when_allread(watched_file)
+        end
         # handlers take care of closing and unwatching
       end
     end
 
+    def common_detach_when_allread(watched_file)
+      watched_file.unwatch
+      watched_file.listener.reading_completed
```

Comment:
@BigYellowHammer this method is undefined in https://github.com/BigYellowHammer/logstash-input-file/blob/feature/terminate-when-EOF/lib/logstash/inputs/file_listener.rb

Reply:
Method `common_detach_when_allread` It's like the method `common_deleted_reaction`, I think we could simply put call `common_deleted_reaction(watched_file, "Completed")` and avoid `common_detach_when_allread`. What do you  think?

Reply:
I changed it, just added the missed callback `reading_completed` method to `LogStash::Inputs::FileListener`

Reply:
I wanted to keep my functionality separate from the main code. Plus `common_deleted_reaction` is more like error handling method when the file gets deleted IMO. In order to merge those methods I will have to put `if` inside `common_deleted_reaction` base on the action and this action is string so it might be error prone to spelling mistakes. So I would rather keep them separate or change string to enum if we really want to merge it.

Reply:
@BigYellowHammer I agree with you, the commit 82e58b054de038e5c6d1e128fcd6f0404ce28152 has fixed that

Reply:
@andsel build failed? Do you think it's intermittent. Can we retrigger?

Reply:
I saw, and restarting make it always fails. I try to reproduce locally

Reply:
@BigYellowHammer the error happens also on master, so it's not related to this PR

Reply:
@andsel do we even need this "an interruptible input plugin" shared example? is it used by some libraries above? because I don't see any direct usage in `logstash-input-file` spec folder